scripts/prepare_fsd50k.py

```diff
@@ -0,0 +1,159 @@
+        self.gen_dataset_splits()
+    
+    def save_fma_to_dcase_json(self):
+        with open(os.path.join(self.metadata_path, 'fma_to_dcase.json'), 'w') as json_file:
+            json.dump(self.fma_to_dcase_dict, json_file, indent=4)
+        
+    def download_dataset(self):
+        os.makedirs(self.dataset_home, exist_ok=True)
+        os.makedirs(self.metadata_path, exist_ok=True)
+        # download fma_small zip
+        download_file(self.base_url, os.path.join(self.dataset_home, self.zip_name))
+        extract_zip(os.path.join(self.dataset_home, self.zip_name), self.dataset_home)
+        print("Done unzipping")
+        # download fma metadata
+        print("Downloading fma_small metadata...")
+        download_file(self.metadata_url, os.path.join(self.metadata_path, self.zip_name))
+        extract_zip(os.path.join(self.metadata_path, self.zip_name), self.metadata_path)
+        print("Done unzipping")
+    
+    def gen_dataset_splits(self):
+        tracks = pd.read_csv(os.path.join(self.metadata_path,'fma_metadata/tracks.csv'), header=[0, 1], index_col=0)
+        genres = tracks['track']['genre_top'].unique()
+        # Loop through the genre 8 classes
+        for genre in genres:
+            # Get tracks by genre, consider only set from "small" tracks
+            genre_tracks = tracks[(tracks['track', 'genre_top'] == genre) & (tracks['set', 'subset'] == 'small')]
+            # Get ntracks_genre from the current genre
+            train_tracks = genre_tracks[:self.ntracks_genre]
+            for track_id, track in train_tracks.iterrows():
+                # get track name by id
+                subdir = f'{track_id:06}'[:3]
+                fma_track_path = os.path.join(self.dataset_home, self.dataset_name, subdir, f'{track_id:06}.mp3')
+                # Based on prob decide test vs. train split
+                save_dir = None
+                # Define DCASE path format variables
+                super_class = "music"
+                fold = "train" if np.random.rand() < self.split_prob else "test"
+                class_name = str(genre)
+                dcase_path = os.path.join(super_class, fold, str(genre), f'{track_id:06}.mp3')
+                # generate dcase to fma map
+                if fold not in self.fma_to_dcase_dict:
+                    self.fma_to_dcase_dict[fold] = {}
+                self.fma_to_dcase_dict[fold][dcase_path] = fma_track_path
+    
+
+class FSD50KDataLoad:
+    def __init__(self,
+                dataset_name="fsd50k",dataset_home=None,
+                download_dataset=False,music_dataset=None,
+                music_home=None,download_music=False,
+                dcase_sound_events_txt="../dcase_datagen/metadata/fsd50k/sound_event_fsd50k_filenames.txt",
+                metadata_path="../dcase_datagen/metadata/fsd50k"):
+        self.dataset_name = dataset_name
+        self.dataset_home = dataset_home
+        self.download_dataset = download_dataset
```

Comment:
this boolean conflicts with the function definition in line 99. One of the two needs to be renamed.

Reply:
Resolved